test/unittest/rtps/history/TopicPayloadPoolRegistryTests.cpp

```diff
@@ -75,12 +75,12 @@ TEST(TopicPayloadPoolRegistryTests, basic_checks)
     EXPECT_FALSE(pool_b2);
 
     // Repeat allocations and check a different pointer is returned
```

Comment:
This test doesn't make sense because relies on CRT defined behavior:

1- The pools were allocated in the first part of the test from heap in chunks of equal size from C++ heap.
2- References to this original chunks are keept in the pool_backup_xx references.
3- The pools are freed and return to the free storage in the C++ heap.
4- The pools are allocated again **but C++ heap may decide to reuse the step 3 freed memory blocks**. Thus we cannot assure that a different pointer is returned.

Note the chunks are allocated in the heap in the call:
```c++
    std::shared_ptr<TopicPayloadPoolProxy> do_get(
            std::shared_ptr<TopicPayloadPoolProxy>& ptr,
            const std::string& topic_name,
            const BasicPoolConfig& config)
    {
        if (!ptr)
        {
            ptr = std::make_shared<TopicPayloadPoolProxy>(topic_name, config);
        }

        return ptr;
    }
```

Reply:
I have changed the test so it checks the number of times a `TopicPayloadPoolProxy` is destroyed. This was what we were truly trying to test.